cms/djangoapps/contentstore/rest_api/v0/views/advanced_settings.py

```diff
@@ -0,0 +1,152 @@
+
+        **Response Values**
+
+        If the request is successful, an HTTP 200 "OK" response is returned.
+
+        The HTTP 200 response contains a single dict that contains keys that
+        are the course's advanced settings. For each setting a dictionary is
+        returned that contains the following fields:
+
+        * **deprecated**: This is true for settings that are deprecated.
+        * **display_name**: This is a friendly name for the setting.
+        * **help**: Contains help text that explains how the setting works.
+        * **value**: Contains the value of the setting. The exact format
+          depends on the setting and is often explained in the ``help`` field
+          above.
+
+        There may be other fields returned by the response.
+
+        **Example Response**
+
+        ```json
+        {
+            "display_name": {
+                "value": "Demonstration Course",
+                "display_name": "Course Display Name",
+                "help": "Enter the name of the course as it should appear in the course list.",
+                "deprecated": false,
+                "hide_on_enabled_publisher": false
+            },
+            "course_edit_method": {
+                "value": "Studio",
+                "display_name": "Course Editor",
+                "help": "Enter the method by which this course is edited (\"XML\" or \"Studio\").",
+                "deprecated": true,
+                "hide_on_enabled_publisher": false
+            },
+            "days_early_for_beta": {
+                "value": null,
+                "display_name": "Days Early for Beta Users",
+                "help": "Enter the number of days before the start date that beta users can access the course.",
+                "deprecated": false,
+                "hide_on_enabled_publisher": false
+            },
+            ...
+        }
+        ```
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_read_access(request.user, course_key):
+            self.permission_denied(request)
+        course_module = modulestore().get_course(course_key)
+        return Response(CourseMetadata.fetch_all(course_module))
+
+    @apidocs.schema(
+        body=CourseAdvancedSettingsSerializer,
+        parameters=[apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID")],
+        responses={
+            200: CourseAdvancedSettingsSerializer,
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists()
+    def patch(self, request: Request, course_id: str):
+        """
+        Update a course's advanced settings.
+
+        **Example Request**
+
+            PATCH /api/contentstore/v0/advanced_settings/{course_id} {
+                "{setting_name}": {
```

Comment:
@xitij2000 Can you please add a detailed description in the PR with examples explaining how different settings related to the discussions configuration can be updated using this endpoint?

Reply:
I think the documentation in the APIs is exactly what you need to use, but I shall clarify with a few examples in the PR description.